integrationTests/chainSimulator/mempool/mempool_test.go

```diff
@@ -1056,6 +1227,366 @@ func Test_Selection_ShouldNotSelectSameTransactionsWithManyTransactionsAndExecut
 	}
 }
 
+func Test_Selection_ProposeEmptyBlocksAndExecutedBlockNotification(t *testing.T) {
+	t.Parallel()
+
+	host := txcachemocks.NewMempoolHostMock()
+	txpool, err := txcache.NewTxCache(txcache.ConfigSourceMe{
+		Name:                        "test",
+		NumChunks:                   16,
+		NumBytesThreshold:           maxNumBytesUpperBound,
+		NumBytesPerSenderThreshold:  maxNumBytesPerSenderUpperBoundTest,
+		CountThreshold:              math.MaxUint32,
+		CountPerSenderThreshold:     math.MaxUint32,
+		EvictionEnabled:             false,
+		NumItemsToPreemptivelyEvict: 1,
+		TxCacheBoundsConfig: config.TxCacheBoundsConfig{
+			MaxNumBytesPerSenderUpperBound: maxNumBytesPerSenderUpperBoundTest,
+			MaxTrackedBlocks:               maxTrackedBlocks,
+		},
+	}, host)
+
+	require.Nil(t, err)
+	require.NotNil(t, txpool)
+
+	// set the number of transactions that we want for each sender
+	numTxsPerSender := 60_000
+	// assure that we have enough balance for fees
+	initialAmount := big.NewInt(int64(numTxsPerSender) * 50_000 * 1_000_000_000)
+
+	// mock the non-virtual selection session
+	senders := []string{"alice"}
+	accounts := map[string]*stateMock.UserAccountStub{
+		"alice": {
+			Balance: initialAmount,
+			Nonce:   0,
+		},
+	}
+
+	selectionSession := txcachemocks.NewSelectionSessionMockWithAccounts(accounts)
+
+	options := holders.NewTxSelectionOptions(
+		10_000_000_000,
+		30_000,
+		int(selectionLoopMaximumDuration.Milliseconds()),
+		10,
+	)
+
+	nonceTracker := newNoncesTracker()
+
+	// create txs and add them to txpool
+	numTxs := numTxsPerSender * len(senders)
+	for i := 0; i < numTxsPerSender; i++ {
```

Comment:
Not sure, do we have some utilities in `integrationTests/chainSimulator/mempool/testutils_test.go` that can help with having these new tests a bit shorter (more concise)? If not, that's all right, we can refactor the tests (make them more accessible) in the future.